using nconf to store random generated private key and deployed contra…

gufmar opened the pull request:
…ct address for future executions in a separate file

Please forgive and correct all grammar and syntax errors.

Commit:
using nconf to store random generated private key and deployed contract address for future executions in a separate file
Please forgive and correct all grammar and syntax errors.

zuzunker commented on kevm-js/kevm.js:
```diff
@@ -24,43 +22,84 @@ const request = require('request-promise-native')
+	  console.log("#  to learn new things and pass on your knowledge to others.             #")
+	  console.log("#                                                                        #")
+	  console.log("##########################################################################")
+  }
```
Nice one! :D

zuzunker commented on kevm-js/kevm.js:
```diff
@@ -24,43 +22,84 @@ const request = require('request-promise-native')
+		}
+	  });
+  } 
+  var res = web3.eth.accounts.wallet.add(account);
```
This is really cool improvement!

zuzunker commented on kevm-js/kevm.js:
```diff
@@ -24,43 +22,84 @@ const request = require('request-promise-native')
   // ******************************
 
-  console.log("Account = " + account.address)
+  //console.log("Account = " + account.address)
```
Why we don't want to print out the account address?

gufmar replied:
no special reason.  simply commented out too much.

I think about introducing 3 options:
1.) a variable for the config-file (default myConfig.json)  So one can work with different accounts and contracts, by simply switching config-file name in the script. 

2.) a standard and optional verbose log level setting

3.) also optionally log to file what was printed to the screen, in case someone wants to understand what steps were executed to finish in a certain behaviour. This would cover and solve also https://github.com/vantuz-subhuman/cardano-testnet-examples/issues/1

All this to have a flexible/solid base for more methods and activities in step 4-6

zuzunker replied:
I would rather then vote for printing the wallet address by default, since people would probably want to copy it to the explorer. Address is a public info and may be **very** useful for later debugging and diagnostics.

About points:

1, 2 - totally agree

3 - For unix-like systems there's already the [`tee`](https://stackoverflow.com/questions/418896/how-to-redirect-output-to-a-file-and-stdout) tool that does exactly this. On windows there also seem to be [a solution for this](https://stackoverflow.com/questions/796476/displaying-windows-command-prompt-output-and-redirecting-it-to-a-file). If there's a nice way to do that from inside the script - then I would probably vote **for** it, but we at least should consider pro\cons and cost\profit of having a functionality for something that is already doable :)

Commit:
added inital setup guide
grouped all configurable variables together
introduced CONFIGURATION variable
introduced log4js for colored messages to stoud + logfile

Commit:
introduced short sleep before calling aftercount value

zuzunker commented on kevm-js/kevm.js:
```diff
@@ -1,74 +1,149 @@
+const FAUCET_INTERVAL = 60000  // to prevent faucet error (too many requests in given amount of time)
+const WAIT_BETWEEN_CALLS = 5000  // to ensure increment request was executed before calling new counter value
+
+const CONFIGURATION = 'test2'  // to work with and switch between different accounts and contracts (config is stored in ./[name].json)
```
The comment here seem to be out of sync with the actual code :)

```
(config is stored in ./[name].json)
```

But really:
```
nconf.use('file', { file: './config_'+CONFIGURATION+'.json' });
```

zuzunker commented on kevm-js/kevm.js:
```diff
@@ -1,74 +1,149 @@
+FIRST: copy and run the following npm install lines into your console to install all all required modules (if not allready installed)
+
+npm install web3
+npm install crypto
```
Btw, the npm page for `crypto` says that it's a deprecated package and should not be installed - https://www.npmjs.com/package/crypto

There's now the same built-in package, so I think the whole script should work as-is without installing it. I didn't install this package yet and script seem to work for me:

![](https://i.imgur.com/V9UkDkr.png)

Commit:
removed crypto as it's a deprecated package
fixed comment's on CONFIGURATION variables
added solc on npm install commands

vantuz-subhuman commented:
This is awesome contribution! Thank you, @gufmar, so much!
